pkg/reconciler/ingress/status.go

```diff
@@ -326,6 +330,16 @@ func (m *StatusProber) listVirtualServicePodIPs(vs *v1alpha3.VirtualService) ([]
 			selector = selector.Add(*requirement)
 		}
 
+		services, err := m.serviceLister.List(selector)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list Service: %v", err)
+		}
+		if len(services) == 0 {
+			// Skip gateways where we cannot find a corresponding service
```

Comment:
We should probably log here and the comment become redundant.
Something like `Skip Gateway %s/%s because it has no corresponding Service`